app/main/download_data.py

```diff
@@ -238,3 +241,82 @@ def process_api_response(query_params: dict) -> tuple:
         return abort(500), f"Unknown content type: {content_type}"
 
     return content_type, file_content
+
+
+def process_async_download(query_params: dict):
+    """process async download request to start the background process.
+    :param query_params: (dict): Query parameters for the API request.
+    """
+    request_url = (
+        Config.DATA_STORE_API_HOST
+        + "/trigger_async_download"
+        + ("?" + urlencode(query_params, doseq=True) if query_params else "")
+    )
+    requests.post(request_url, timeout=20)
```

Comment:
@sc15zs has merged a change to this endpoint - it is a post call, as we do here, but query_params needs to be passed through the body. If you rebase this branch onto main, you should be able to get the correct changes.

Let me know if you want a second pair of eyes to help get this done. It might be tricky because I can see you've been merging other branches into this one.